samples/scripts/externalclient-setup.js

```diff
@@ -0,0 +1,298 @@
+// ------------------------------
+// For testing AWS credentials, the following are needed:
+// 1. An AWS developer account is needed. The account ID will need to
+//    be provided in the configuration object below.
+// 2. A role for web identity federation. This will also need to be provided
+//    in the configuration object below.
+//    - An OIDC Google identity provider needs to be created with the following:
+//      issuer: accounts.google.com
+//      audience: Use the client_id of the service account.
+//    - A role for OIDC web identity federation is needed with the created
+//      Google provider as a trusted entity:
+//      "accounts.google.com:aud": "$CLIENT_ID"
+//    The role creation steps are documented at:
+//    https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-idp_oidc.html
+//
+// This script needs to be run once. It will do the following:
+// 1. Create a random workload identity pool.
+// 2. Create a random OIDC provider in that pool which uses the
+//    accounts.google.com as the issuer and the default STS audience as the
+//    allowed audience. This audience will be validated on STS token exchange.
+// 3. Enable OIDC tokens generated by the current service account to impersonate
+//    the service account. (Identified by the OIDC token sub field which is the
+//    service account client ID).
+// 4. Create a random AWS provider in that pool which uses the provided AWS
+//    account ID.
+// 5. Enable AWS provider to impersonate the service account. (Principal is
+//    identified by the AWS role name).
+// 6. Print out the STS audience fields associated with the created providers
+//    after the setup completes successfully so that they can be used in the
+//    tests. These will be copied and used as the global AUDIENCE_OIDC and
+//    AUDIENCE_AWS constants in samples/test/externalclient.test.js.
+//    An additional AWS_ROLE_ARN field will be printed out and also needs
+//    to be copied to the test file. This will be used as the AWS role for
+//    AssumeRoleWithWebIdentity when federating from GCP to AWS.
+// The same service account used for this setup script should be used for
+// the test script.
+//
+// It is safe to run the setup script again. A new pool is created and new
+// audiences are printed. If run multiple times, it is advisable to delete
+// unused pools. Note that deleted pools are soft deleted and may remain for
+// a while before they are completely deleted. The old pool ID cannot be used
+// in the meantime.
+
+const fs = require('fs');
+const {promisify} = require('util');
+const {GoogleAuth} = require('google-auth-library');
+
+const readFile = promisify(fs.readFile);
+
+/**
+ * Generates a random string of the specified length, optionally using the
+ * specified alphabet.
+ *
+ * @param {number} length The length of the string to generate.
+ * @return {string} A random string of the provided length.
+ */
+function generateRandomString(length) {
```

Comment:
If this is something that users will be supposed to do every time, maybe instead make it a part of the crypto class and export from there? We don't want this function to be copy-pasted to all of the users' code.

Reply:
This is not meant for public consumption. I don't think we should expose it. It is just a helper function for testing in the following:
1. This script which is meant to configure a project with workload identity pools so integration tests can be run on it. (this is a one time operation).
2. The `samples/test/externalclient.test.js` to facilitate generating random file names needed during the tests.

Reply:
Sounds good!